v2/internal/pkg/registriesd/registriesd.go

```
@@ -138,38 +148,69 @@ func createRegistryConfigFile(registryFileAbsPath, registryHost string) error {
 	return nil
 }
 
-func updateRegistryConfigFile(registryFileAbsPath, registryHost string) error {
-	configFileBytes, err := os.ReadFile(registryFileAbsPath)
-	if err != nil {
-		return fmt.Errorf("error reading registry config file %w", err)
-	}
+func loadRegistryConfigFiles(customizableRegistriesDir string) []registryConfiguration {
+	var configFiles []registryConfiguration
 
-	var registryConfigStruct registryConfiguration
-	err = yaml.Unmarshal(configFileBytes, &registryConfigStruct)
+	files, err := os.ReadDir(customizableRegistriesDir)
 	if err != nil {
```

@aguidirh - what is the impact here if we do have ReadDir failure , why mask the error ?

---

The goal of `loadRegistryConfigFiles` is only to check if there are config files already in place so oc-mirror does not need to generate them, so in case ReadDir fail, it won't impact anything, it will only go ahead and create the mandatory config files.

---

Basically the goal is `try to load config files, if there are no config files, only return an empty array`

---

Hi @lmzuccarelli,

As I explained before, if no files were found using `ReadDir` we just ignore the error, but on the other hand if files were found but they failed to unmarshal, then we handle the error. 

This was suggested by @r4f4, I implemented it [here](https://github.com/openshift/oc-mirror/compare/9e1bc8b9d9c65b3fb5a7f7b9c81d06d1442159ef..f8bab227815a6bf7953b35e81b71bf759ccb48bc))